daemon/restart.go

```diff
@@ -39,8 +39,10 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		return err
 	}
 
-	if err := daemon.containerStart(container); err != nil {
-		return err
+	if !container.HostConfig.AutoRemove {
```

Comment:
But if you don't add this if-statement then I think it'll act the way the end user expects.  There's a difference between "docker stop" and "docker restart".  Stop is asking for the container to fully stop running.  Restart is just asking for it to be rebooted NOT stopped - not unlike a daemon reboot would.  If we don't do this then there's no difference between "docker stop" and "docker restart" - and I don't think that's correct. A restart is the user's way of explicitly telling us that they don't want the container to be fully stopped, otherwise they would just use "stop".